### ISSUE TITLE
Variables from controllers/base.php not accessible in template

### ISSUE THREAD

**Issue by slackday** [soberwp/controller#14]
I'm using Sage 9.0.0-beta.2 and also the compatible version of this controller. Which I really like a lot btw :)

Reading the documentation about global variables I'm experiencing an issue.

I have created a simple `base.php`controller which looks like following:

```
<?php

namespace App;

use Sober\Controller\Controller;

class Base extends Controller
{
    public function footer()
    {
        $footer = [];

        if (function_exists('get_field'))
        {
          $footer['text'] = get_field('footer_text', 'options');
        }

        return $footer;
    }
}
```
I also have controller `controllers/page.php` which runs.

If I `var_dump` inside the controller I can see that the code is running. However the variable is not present in `templates/partials/footer.blade.php`.

Why are variables in `base` not present when `page` also runs?

There are many workarounds if I use `traits` it works but I have to include the trait in Page which seems like overhead since I plan to add more controllers.

The best workaround is to add `protected $tree = true; // Inherit all templates` to Page but this is the same overhead as above.

Also I would find it useful if there was an options like "reverese tree" instead of using `class X extends Controller implements Tree` it would be cool if there was an global or "must use" kind of like Base but you could call the controller what you want and it would always run.

Thanks!

**Comment by darrenjacoby** [comment]
Hi @slackday thanks for the feedback.  It sounds like a bug to be honest, base should be included in all files.  I'll try to recreate on my side and then revert as soon as I can.

**Comment by darrenjacoby** [comment]
@slackday I've tried to recreate this on my end, but it's working as should.  The variables from `src/controllers/Base.php` are useable in `templates/partials/footer.blade.php`. 

Could you perhaps provide me with more information/code to try debug?  Can you show me the templates code?

**Comment by vdrnn** [comment]
@darrenjacoby @slackday I tested this a bit further and figured out that there is a problem with camelCase function names. So if my function in the controller is named `foobar` and I call it in the template `$foobar` everything works. If I name the function `foobarTest` and call `$foobarTest` I get the following error: 

> Fatal error: Uncaught Symfony\Component\Debug\Exception\FatalThrowableError: Call to a member function have_posts() on

**Comment by darrenjacoby** [comment]
@vdrnn the camelCase function name is converted to snake_case in the view.  This is the intended behavior as snake case is commonly used in PHP.  If you run @debug in the view it will output the correct variable name to use.

**Comment by vdrnn** [comment]
@darrenjacoby bam! that was it. Thanks! 👍

**Comment by darrenjacoby** [comment]
@vdrnn no problem!

**Comment by slackday** [comment]
I'm not sure how I would create example. Do you mean setup a separate repo with code example?

What I've done is install sage theme 9.0.0-beta2

Install this plugin beta 9.0.0-beta2 as instructions in readme.

Create folder in sage theme called `controllers`. Add two controller files `base.php` `page.php`. (lower case but this seems to work the same independant of casing)

`controllers/base.php`
```
<?php

namespace App;

use Sober\Controller\Controller;

class Base extends Controller
{
    // Global

    public function global() {
      $global = array();

      $page_for_posts = get_option('page_for_posts');
      $page_for_posts = get_post($page_for_posts);
      $global['page_for_posts'] = $page_for_posts->ID;

      return $global;
    }


    // Footer

    public function page_foot()
    {
        $footer = [];

        if (function_exists('get_field'))
        {
          $footer['text'] = get_field('footer_text', 'options');
        }

        return $footer;
    }
}
```

`controllers/page.php`
```
<?php

namespace App;

use Sober\Controller\Controller;

class Page extends Controller
{
    // Traits

    use Modules;
    use Sidebar;


    // Inherit all templates

    protected $tree = true;


    // Variables

    public function lead()
    {
        $lead = null;

        if (function_exists('get_field'))
        {
            $lead = get_field('page_lead_text');
        }

        return $lead;
    }
}
```

Variables are accessible on page in footer template if I have `protected $tree = true;` if i comment out `protected $tree = true;` variables disappear.

**Comment by darrenjacoby** [comment]
@slackday 

I've looked into this and you are right, there is a bug in the code.  I'm in the process of fixing and will push a new version by tomorrow.  

This issue applies to both beta2 and master so thanks for picking it up.

**Comment by darrenjacoby** [comment]
Fixed, you can upgrade to 9.0.0-beta2.1 and this should now work.
